tests/reconfiguration.py

```diff
@@ -56,14 +60,25 @@ def test_add_node(network, args):
 @reqs.description("Adding a valid node from a backup")
 @reqs.at_least_n_nodes(2)
 def test_add_node_from_backup(network, args):
-    backup = network.find_any_backup()
+    primary, backup = network.find_primary_and_any_backup()
+
+    # Retrieve snapshot from primary as only primary node
+    # generates snapshots
+    snapshot_dir = network.get_committed_snapshots(primary)
+
     new_node = network.create_and_trust_node(
-        args.package, "local://localhost", args, target_node=backup
+        args.package,
+        "local://localhost",
+        args,
+        target_node=backup,
+        snapshot_dir=snapshot_dir,
     )
     assert new_node
     return network
 
 
+# Note: this test cannot be included in the test suite yet as
+# add_from_snapshot() decorator makes use of historical queries
```

Comment:
Link to #1986?

Reply:
This is actually #1648 but yes!

Reply:
Indeed, definitely a good thing to link then :)